src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -1425,12 +1427,20 @@ def forward(
 
         # Cross-Attention Text
         queries = self.with_pos_embed(hidden_states, position_embeddings)
+        text_encoder_attention_mask = text_encoder_attention_mask[:, None, None, :]
+        text_encoder_attention_mask = text_encoder_attention_mask.repeat(
+            1, self.num_attention_heads, self.num_queries, 1
+        )
+
+        dtype = torch.float16
```

Comment:
We shouldn't explicitly set the dtype here like this as it will break with other assumptions in the library about weight loading e.g. I should be able to do `model = AutoModel(checkpoint, torch_dtype=torch.float32)` without issue. I'd expect the dtype to be taken from e.g. the text_encoder_hidden_states or the dtype of the module - self.dtype

Reply:
Ops, this was a copy-pasta! Thanks for spotting this. Fixed here https://github.com/huggingface/transformers/pull/30364/commits/539763d21cb6469267ef7b54bdbf085d5c90c7b1 for the cross-attention and for the text enhancer that had the same logic